### ISSUE TITLE
Django 5 compatibility

### ISSUE THREAD

**Issue by jakubste** [bikeshedder/django-composite-field#10]
Hey!

I'm trying to upgrade my project to Django 5 and I run into such an issue:

```
  File "/Users/jakubste/.virtualenvs/bdsm-S0d35qI4-py3.11/lib/python3.11/site-packages/django/db/models/base.py", line 511, in __init__
    if field.attname not in kwargs and field.column is None or field.generated:
                                                               ^^^^^^^^^^^^^^^
AttributeError: 'PartialDateField' object has no attribute 'generated'
```

`PartialDateField` is my class that uses `CompositeField` and it's quite simple:

```python
class PartialDateField(CompositeField):
    year = models.IntegerField(null=True, blank=True)
    month = models.IntegerField(null=True, blank=True)
    day = models.IntegerField(null=True, blank=True)
```

Django 5 introduced generated fields and it looks like the fix is as easy as adding `generated = False` field in `CompositeField`.

Any chances for patching this? :)

**Comment by bikeshedder** [comment]
Could you create a PR for that?

Bonus points for updating the tox config and making sure it doesn't break older Django versions. :smiley:

**Comment by jakubste** [comment]
https://github.com/bikeshedder/django-composite-field/pull/11

I have zero experience with tox configs, so for now just a PR for Django 5

**Comment by bikeshedder** [comment]
I merged #11.

I'll release 2.0 tomorrow. I want to convert the project to poetry (see #7) first as the current way of creating releases is a bit wonky and very outdated. I'll drop support for all outdated Python and Django versions while I'm at it and will check if tox is still the way to go. It's been quite some time since I last looked into testing code for multiple library versions.

**Comment by bikeshedder** [comment]
This was fixed by merging:
- #11

**Comment by bikeshedder** [comment]
I just released 2.0.0 on PyPI: https://pypi.org/project/django-composite-field/

Thanks for the heads up.
